src/flows/parser_item_html_backfill_filesplit_flow.py

```diff
@@ -79,31 +57,22 @@ def stage_chunk(key: str, index: int, df: pd.DataFrame) -> str:
     return stage_key
 
 
-def stage(key_dfs: (str, List[pd.DataFrame])) -> [str]:
-    logger = prefect.context.get("logger")
-    key, dfs = key_dfs
-    logger.info(f"Number of dfs: {len(dfs)}.")
-    df = pd.concat(dfs)
-    chunks = [(i, df[i:i + STAGE_CHUNK_ROWS]) for i in range(0, df.shape[0], STAGE_CHUNK_ROWS)]
-    logger.info(f"Number of chunks created: {len(chunks)}.")
-    keys = [stage_chunk(key, i, chunk) for i, chunk in chunks]
-    logger.info(f"Staged keys: {*keys,}")
-    return keys
-
 @task()
-def split_files_process(key: str):
+def transform_stage_process(df_chunk: (int, pd.DataFrame), key: str):
     """
     Split S3 files into smaller stage files
     """
-    for index, df in enumerate(split_file(key)):
-        df_transformed = transform(df)
-        stage_chunk(key=key, index=index, df=df_transformed)
+    index, df = df_chunk
+    df_transformed = transform(df)
+    stage_chunk(key=key, index=index, df=df_transformed)
 
 
 with Flow(FLOW_NAME, executor=LocalDaskExecutor(scheduler="processes", num_workers=50)) as flow:
-    keys = get_source_keys()
-    split_files_process.map(keys)
+    key = Parameter('key', required=True)
+    df_chunks = split_file(key)
+    transform_stage_process.map(df_chunks, key=unmapped(key))
```

Comment:
I think you want to keep these steps as a single task without parallelism, to avoid using up all memory.